4-layers/Dockerfile.better

```diff
@@ -0,0 +1,25 @@
+FROM ruby:alpine
+
+RUN apk update && \
+    apk add --no-cache alpine-sdk libmagic \
+    nodejs npm postgresql-dev \
+    chromium aws-cli jq && \
+    rm -rf /var/cache/apk/* && \
+    apk -U upgrade
+
+RUN npm install -g npm@9 serverless@3
+
+ENV GEM_HOME="/usr/local/bundle"
+ENV PATH=$GEM_HOME/bin:$GEM_HOME/gems/bin:$PATH
+
+# Install gems
+COPY Gemfile ./
+COPY Gemfile.lock ./
+RUN bundle install
+
+# selectively copy the tests to the image's working directory
+COPY tests/ /tests/
```

Comment:
This folder is much more likely to change than the following two lines, consider even the order of these to get more gains.